src/Symfony/Component/Notifier/Bridge/Sweego/SweegoTransport.php

```diff
@@ -0,0 +1,141 @@
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage
+            && (null === $message->getOptions() || $message->getOptions() instanceof SweegoOptions);
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $body = [
+            'recipients' => [
+                [
+                    'num' => $message->getPhone(),
+                    'region' => $this->region,
+                ],
+            ],
+            'message-txt' => $message->getSubject(),
+            'channel' => 'sms',
+            'campaign-type' => $this->campaignType,
+            'provider' => 'sweego',
+        ];
+
+        $options = $message->getOptions()?->toArray() ?? [];
+        $body = $this->setBat($body, $options);
+        $body = $this->setCampaignId($body, $options);
+        $body = $this->setShortenUrls($body, $options);
+        $body = $this->setShortenWithProtocol($body, $options);
+
+        $endpoint = \sprintf('https://%s/send', $this->getEndpoint());
+        $response = $this->client->request('POST', $endpoint, [
+            'headers' => [
+                'Api-Key' => $this->apiKey,
+            ],
+            'json' => $body,
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote Sweego server.', $response, 0, $e);
+        }
+
+        if (200 !== $statusCode) {
+            throw new TransportException('Unable to send the SMS.', $response);
+        }
+
+        $success = $response->toArray(false);
+
+        $sentMessage = new SentMessage($message, (string) $this);
+        $sentMessage->setMessageId(array_values($success['swg_uids'])[0]);
+
+        return $sentMessage;
+    }
+
+    private function setBat(array $body, array $options): array
+    {
+        $body['bat'] = (bool) ($options[SweegoOptions::BAT] ?? false);
+
+        return $body;
+    }
+
+    private function setCampaignId(array $body, array $options): array
+    {
+        if (!array_key_exists(SweegoOptions::CAMPAIGN_ID, $options) && \is_string($options[SweegoOptions::CAMPAIGN_ID])) {
+            $body['campaign-id'] = $options[SweegoOptions::CAMPAIGN_ID];
+        }
+
+        return $body;
+    }
+
+    private function setShortenUrls(array $body, array $options): array
+    {
+        $body['shorten_urls'] = (bool) ($options[SweegoOptions::SHORTEN_URLS] ?? false);
+
+        return $body;
+    }
+
+    private function setShortenWithProtocol(array $body, array $options): array
+    {
+        $body['shorten_with_protocol'] = (bool) ($options[SweegoOptions::SHORTEN_WITH_PROTOCOL] ?? false);
+
+        return $body;
+    }
```

Comment:
In other bridges we are setting these values via the constructor. We don't use an `$options` array

Reply:
I took the example on https://github.com/symfony/symfony/blob/7.2/src/Symfony/Component/Notifier/Bridge/Lox24/Lox24Options.php

I can fix Sweego one to make it similar to other bridges, and fix Lox24 on another PR. WDYT?

Reply:
`SmsmodeOptions`, `SmsboxOptions` and some others also expose kind of setters. In which did you figured out setting values via the constructor?
I'm agree to harmonize it, let's discuss this topic on a dedicated issue?

Reply:
TO clarify here, I am talking about the constructor of the transport. For the Options classes, I don't have a preference.

To my understanding, all options, which could differ from message to message should only provided via Options class. In case where we have a default region for example and you can set this region also via Options class, it should be merge in the transport. But you implemented setters on the transport to set options and also have `private readonly array $options = [],` on the transport.

I would change that to:
```diff
    public function __construct(
        #[\SensitiveParameter] private readonly string $apiKey,
        private readonly string $region,
        private readonly string $campaignType,
-        private readonly array $options = [],
+        private readonly bool $bat = false,
+.       // ....
        ?HttpClientInterface $client = null,
        ?EventDispatcherInterface $dispatcher = null,
    ) {
        parent::__construct($client, $dispatcher);
    }
```

I hope its more clear now